spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -42,10 +45,41 @@
 
 	private final CredentialsProvider credentialsProvider;
 
+	private final HeaderProvider headerProvider;
+
+	private final ExecutorProvider systemExecutorProvider;
+
+	private final FlowControlSettings flowControlSettings;
+
+	private final Duration maxAckDurationPeriod;
+
+	private final Integer parallelPullCount;
+
+	/**
+	 * Base {@link DefaultSubscriberFactory} constructor.
+	 *
+	 * @param projectIdProvider provides the GCP project ID
+	 * @param executorProvider provides the executor to be used by the subscriber, useful to specify
+	 *                         the number of threads to be used by each executor
+	 * @param channelProvider provides the channel to be used by the subscriber
+	 * @param credentialsProvider provides the GCP credentials to be used by the subscriber on the
+	 *                            API calls it makes
+	 * @param headerProvider provides headers for the API's HTTP calls
+	 * @param systemExecutorProvider provides an executor for polling and managing lease extensions
+	 * @param flowControlSettings configures the flow control of the subscriber, including the
+	 *                            behaviour for when flow limits are hit
+	 * @param maxAckDurationPeriod sets the maximum period the ack period is extended by
+	 * @param parallelPullCount sets the number of pull workers
+	 */
 	public DefaultSubscriberFactory(GcpProjectIdProvider projectIdProvider,
 			ExecutorProvider executorProvider,
 			TransportChannelProvider channelProvider,
-			CredentialsProvider credentialsProvider) {
+			CredentialsProvider credentialsProvider,
+			HeaderProvider headerProvider,
+			ExecutorProvider systemExecutorProvider,
+			FlowControlSettings flowControlSettings,
+			Duration maxAckDurationPeriod,
+			Integer parallelPullCount) {
```

Comment:
No, that's not what I'd like to see for end-user API. That is really confusing. Would be better to move non-mandatory properties to setters. If you worry about autowiring, we have a Spring Boot auto-configuration on the matter.

More over no one is going to provide `Duration` and `Integer` beans.

Out of topic: `Duration` is a Java 8 class. So, end-users won't see benefits to stick with SF-4.3 when there is already SF-5.0 for better Java 8 support. I mean end-users might start complain why we require Java 8 if we don't support SF-5.0? Or: why we are not based on SF-5.0 if we require Java 8?

:shrugged:

Reply:
We're using `org.threeten.bp.Duration`, a backport for Java 5, 6 and 7. Although I have no issues requiring Java 8.

On the topic of SI 5.0, the way we're currently controlling our SI version is by setting `spring-cloud-gcp` `pom.xml`'s parent to `spring-cloud-build`.
Changing its version to the latest `2.0.0.M6` would always change SI to 5 and SB to 2.

In other words, is there a way to change SI to 5 while keeping SB < 2? Is that even recommended?
We could always explicitly set an SI version in our bom or parent pom to 5...

Reply:
The story is not about SI, but SF. Anyway I agree that it's not good to pull those version if other aren't bumped. I still think that we have to rely on the Spring Platform management dependencies as much as possible (Spring Cloud build should do that for us) and only use explicit versions for Spring Projects if they have some critical fix.